### PR TITLE
fix upgrade restore workflow

### PR DESCRIPTION
**Pull request by jdsn**
The upgrade workflow with an upgrade differs from the one without upgrade (backup and restore on the same cloud release).

This PR implements the restore for the upgrade case.

### CODE REVIEW

**Comment by bmwiedemann** [comment]
+1

**Comment by jdsn** [comment]
> why the shift?

just a good habit :)  but I agree, its not needed

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -4054,10 +4053,24 @@ function onadmin_crowbarrestore()
...
         wait_for 20 10 "onadmin_is_crowbar_api_available" "crowbar service to start"
-        # crowbarctl needs --anonymous to workaround a crowbarctl issue which leads to two api requsts
-        # per call (auth + actual request) which fails when running crowbarctl directly on the admin node
-        safely crowbarctl backup upload /tmp/$btarball --anonymous
-        safely crowbarctl backup restore $btarballname --anonymous --yes
+        case $restoremode in
+            with_upgrade)
+                # restore after upgrade has different workflow (missing APIs) than
+                #   a restore from a backup of the same cloud release
+                local http_code=`curl -X POST -F "file=@/tmp/$btarball" -s -o upgrade-start.txt -w "%{http_code}" $crowbar_api/installer/upgrade/start`
```
Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: I guess you can't write-out the `http_code` here as long as it is not defined yet

**Comment by MaximilianMeister** [comment]
the code looks good to me
+1 after addressing the `http_code` variable

**Pull request merged**